crates/hotshot/src/traits/election/static_committee_leader_two_views.rs

```diff
@@ -1,202 +1,159 @@
-        let entry = pub_key.stake_table_entry(1u64);
-        if self.committee_nodes_with_stake.contains(&entry) {
-            Some(entry)
-        } else {
-            None
-        }
+        // Only return the stake if it is above zero
+        self.indexed_stake_table.get(pub_key).cloned()
     }
 
-    fn create_election(
-        mut all_nodes: Vec<PeerConfig<PUBKEY>>,
-        committee_members: Vec<PeerConfig<PUBKEY>>,
-        committee_topic: Topic,
-        fixed_leader_for_gpuvid: usize,
-    ) -> Self {
-        let mut committee_nodes_with_stake = Vec::new();
-        let mut committee_nodes_without_stake = Vec::new();
-
-        // Iterate over committee members
-        for entry in committee_members
-            .iter()
-            .map(|entry| entry.stake_table_entry.clone())
-        {
-            if entry.stake() > U256::from(0) {
-                // Positive stake
-                committee_nodes_with_stake.push(entry);
-            } else {
-                // Zero stake
-                committee_nodes_without_stake.push(PUBKEY::public_key(&entry));
-            }
-        }
-
-        // Retain all nodes with stake
-        all_nodes.retain(|entry| entry.stake_table_entry.stake() > U256::from(0));
+    /// Check if a node has stake in the committee
+    fn has_stake(&self, pub_key: &<TYPES as NodeType>::SignatureKey) -> bool {
+        self.indexed_stake_table
+            .get(pub_key)
+            .is_some_and(|x| x.stake() > U256::zero())
+    }
 
-        debug!(
-            "Election Membership Size: {}",
-            committee_nodes_with_stake.len()
-        );
+    /// Get the network topic for the committee
+    fn get_committee_topic(&self) -> Topic {
+        self.committee_topic.clone()
+    }
 
-        Self {
-            all_nodes_with_stake: all_nodes
-                .into_iter()
-                .map(|entry| entry.stake_table_entry)
-                .collect(),
-            committee_nodes_with_stake,
-            committee_nodes_without_stake,
-            fixed_leader_for_gpuvid,
-            _type_phantom: PhantomData,
-            committee_topic,
-        }
+    /// Index the vector of public keys with the current view number
+    fn get_leader(&self, view_number: TYPES::Time) -> TYPES::SignatureKey {
+        let index =
+            usize::try_from((*view_number / 2) % self.eligible_leaders.len() as u64).unwrap();
+        let res = self.eligible_leaders[index].clone();
+        TYPES::SignatureKey::public_key(&res)
     }
 
+    /// Get the total number of nodes in the committee
     fn total_nodes(&self) -> usize {
-        self.committee_nodes_with_stake.len()
+        self.stake_table.len()
     }
 
+    /// Get the voting success threshold for the committee
     fn success_threshold(&self) -> NonZeroU64 {
-        NonZeroU64::new(((self.committee_nodes_with_stake.len() as u64 * 2) / 3) + 1).unwrap()
+        NonZeroU64::new(((self.stake_table.len() as u64 * 2) / 3) + 1).unwrap()
```

Comment:
Looks like you didn't add them originally, but since it's touched here I might ask: what are these magic numbers for? And can we perhaps do named constants instead? I feel like if I had to change this code I'd end up spending 20 minutes just working out what these mean

Reply:
Idk what I would name the numbers here, it's just 2/3 + 1